dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -1047,14 +1047,19 @@ private void checkNotHosted(Tool tool) {
     public Response getToolZip(@ApiParam(hidden = true) @Auth Optional<User> user,
         @ApiParam(value = "toolId", required = true) @PathParam("toolId") Long toolId,
         @ApiParam(value = "tagId", required = true) @PathParam("tagId") Long tagId) {
-        Tool tool;
-        if (user.isPresent()) {
-            tool = toolDAO.findById(toolId);
+
+        Tool tool = toolDAO.findById(toolId);
+        if (tool.getIsPublished()) {
+            tool = toolDAO.findPublishedById(toolId);
```

Comment:
This structure seems odd. If you have it already and know its published, why go and get it a second time by id?